components/new-contribution-flow/StepDetails.js

```diff
@@ -115,13 +115,13 @@ const StepDetails = ({ onChange, data, collective, tier, showFeesOnTop, isTaxDed
           </P>
         </React.Fragment>
       )}
-      {tier.customFields && (
+      {tier?.customFields && (
         <Box mt={28}>
           <H5 fontSize="20px" fontWeight="normal" color="black.800">
             <FormattedMessage id="OtherInfo" defaultMessage="Other information" />
           </H5>
           <TierCustomFields
-            fields={tier.customFields}
+            fields={tier?.customFields}
```

Comment:
Shouldn't be necessary, if the tier doesn't exist we won't enter the block (`tier?.customFields &&`)

```suggestion
            fields={tier.customFields}
```